Multisig/UI/Transactions/TransactionsViewModel.swift

```diff
@@ -0,0 +1,428 @@
+//
+//  TransactionListViewModel.swift
+//  Multisig
+//
+//  Created by Moaaz on 5/27/20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import Combine
+import SwiftCryptoTokenFormatter
+import BigInt
+
+struct TransactionsList {
+    struct Section {
+        var name: String
+        var transactions: [BaseTransactionViewModel]
+
+        var isEmpty: Bool {
+            transactions.isEmpty
+        }
+    }
+
+    var sections: [Section] = []
+
+    var isEmpty: Bool {
+        sections.allSatisfy { $0.isEmpty }
+    }
+}
+
+class BaseTransactionViewModel {
+
+    var nonce: String?
+    var status: TransactionStatus
+    var date: Date?
+    var formattedDate: String
+    var confirmationCount: Int?
+    var threshold: Int?
+
+    init() {
+        status = .success
+        formattedDate = ""
+        date = Date()
+    }
+}
+
+
+class TransferTransaction: BaseTransactionViewModel {
+
+    var address: String
+    var isOutgoing: Bool
+    var amount: String
+    var tokenSymbol: String
+
+    override init() {
+        amount = "0"
+        tokenSymbol = "ETH"
+        isOutgoing = true
+        address = ""
+        super.init()
+    }
+}
+
+
+class ChangeMasterCopyTransaction: BaseTransactionViewModel {
+
+    var contractVersion: String
+    var contractAddress: String
+
+    override init() {
+        contractVersion = ""
+        contractAddress = ""
+        super.init()
+    }
+}
+
+
+class SettingChangeTransaction: BaseTransactionViewModel {
+
+    var title: String
+
+    override init() {
+        title = ""
+    }
+
+}
+
+class CustomTransaction: TransferTransaction {
+    var dataLength: Int
+
+    override init() {
+        dataLength  = 98
+        super.init()
+    }
+}
+
+extension TransactionStatus {
+    static let queueStatuses = [TransactionStatus.pending, .waitingConfirmation, .waitingExecution]
+    static let historyStatuses = [TransactionStatus.success, .failed, .canceled]
+
+    var isInQueue: Bool {
+        Self.queueStatuses.contains(self)
+    }
+
+    var isInHistory: Bool {
+        Self.historyStatuses.contains(self)
+    }
+}
+
+
+class TransactionsViewModel: ObservableObject {
```

Comment:
we should create a tech debt task to create unit tests for this view model. And for other mentioned places too.